hooks/tk-multi-publish2/maya/publish_files_integ.py

```diff
@@ -0,0 +1,266 @@
+        names = ""
+        extras = list(set(groups) - set(group_nodes))
+        # check for any group nodes apart from the once mentioned
+        if extras:
+            for item in extras:
+                names += "\n" + item
+            self.logger.error("Incorrect naming or extra group nodes present in the scene.",
+                              extra={
+                                  "action_show_more_info": {
+                                      "label": "Show Info",
+                                      "tooltip": "Show the conflicting group nodes",
+                                      "text": "Please check the group nodes:\n{}".format(names) +
+                                              "\n\nOnly following group nodes should be present:\nWORLDSCALE \nSET_TO_WORLD \nTRACK_GEO"
+                                  }
+                              }
+                              )
+            return False
+        # check if any of the required group nodes are missing
+        elif not set(group_nodes).issubset(set(groups)):
+            self.logger.error("Please ensure all the group nodes are present.",
+                              extra={
+                                  "action_show_more_info": {
+                                      "label": "Show Info",
+                                      "tooltip": "Group nodes",
+                                      "text": "Following group nodes should be present:\nWORLDSCALE \nSET_TO_WORLD \nTRACK_GEO"
+                                  }
+                              }
+                              )
+            return False
+        return self._check_hierarchy(group_nodes)
+
+    @staticmethod
+    def _is_group(node=None):
+        """Check for all the group nodes present in the scene.
+            :param:
+                node: all the nodes in the scene
+        """
+        if cmds.nodeType(node) != "transform":
+            return False
+
+        children = cmds.listRelatives(node, c=True)
+        if not children:
+            return True
+
+        for c in children:
+            if cmds.nodeType(c) != 'transform':
+                return False
+        else:
+            return True
+
+    def validate(self, task_settings, item):
+        """
+        Validates the given item to check that it is ok to publish. Returns a
+        boolean to indicate validity.
+
+        :param task_settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        :returns: True if item is valid, False otherwise.
+        """
+        print item.properties
+        all_dag_nodes = cmds.ls(dag=True, sn=True)
+        groups = [g for g in all_dag_nodes if self._is_group(g)]
+        status = True
+        status = self._node_naming(_group_nodes, groups) and status
```

Comment:
If this is False, it should error before going to the next validation method, right?

Reply:
This will go through all the validation methods being called in the main validation() and return a cumulative result of all of that. It is so that all the possible error messages could be shown together.

I am calling group node validation method and camera validation method in validation() so that if validations fail for any or both of them, the artists are able to see the errors simultaneously.
The other validations are dependent on these two and will give the error message once these pass the check.